### DISCUSSION TITLE
Nested Properties

### DISCUSSION THREAD

**Discussion by ArneNostitz** [chrisgurney/obsidian-note-toolbar#74] [Ideas]
Hi, trying to get props from nested YAML. is this possible?

email:
  - email: eddi@nono.com

URI: mailto:{{prop_email[0].email}}

but i am not able to :) tried all kinds of different versions, but with no success. any tipps? thanks!

**Comment by chrisgurney** [comment]
Nested properties are [not supported](https://forum.obsidian.md/t/cannot-edit-nested-properties-in-preview-mode/64413) in Obsidian, so I don't have an API I can easily use to do this.

For those, like yourself, who would still like to use nested properties, I've seen them used quite differently -- see [this thread](https://forum.obsidian.md/t/properties-support-multi-level-yaml-nested-attributes/63826/12) for a number of examples.

If there's a way I can do this, ideally by _not_ including another library, and provide a variable syntax that works across the various ways people use it, it's possible, but I think I'd need more upvotes on this particular enhancement request.

**Reply by ArneNostitz** [reply]
i see, yes the "workarounds" of nested yaml are quite something. i use meta-bind to do what im trying to do ... 
dataview, JS Engine and meta-bind as well as meta data menu all use the same structure to access them though. like you see in my first posting - its all an array. 

Key[array position starting from 0].sub-key

**Reply by chrisgurney** [reply]
@ArneNostitz I may just end up looking into implementing Dataview support then, if that also solves your use case.

Thanks for your insight!

**Reply by ArneNostitz** [reply]
i guess thats generally a good idea :) but look into it - i´ve read a few months ago that they rewrite dataview (?) ...

**Reply by ArneNostitz** [reply]
another way of accessing these is by using "`=this.key`" ... so i dont even know if dataview is needed - but i am not a programmer - so all my knowledge is super dangerous "half knowledge"

**Reply by chrisgurney** [reply]
> i guess thats generally a good idea :) but look into it - i´ve read a few months ago that they rewrite dataview (?) ... 

Yeah, there's [Datacore](https://github.com/blacksmithgu/datacore), and something on the [Obsidian roadmap](https://obsidian.md/roadmap/) called "Dynamic views" that might replace Dataview, but it's anyone's guess at the moment.

I made a previous attempt but didn't quite get there. If the effort and overhead is minimal, I think it's worth another shot.

**Reply by chrisgurney** [reply]
Regarding Dataview support, see: #64
